apps/fz_http/lib/fz_http/events.ex

```diff
@@ -3,91 +3,75 @@ defmodule FzHttp.Events do
   Handles interfacing with other processes in the system.
   """
 
-  alias FzHttp.{Devices, Rules, Users, Notifications}
+  alias FzHttp.{
+    Users,
+    Notifications,
+    AllowRules,
+    Gateways,
+    Devices,
+    Devices.Device,
+    AllowRules.AllowRule
+  }
 
   require Logger
 
   # set_config is used because devices need to be re-evaluated in case a
   # device is added to a User that's not active.
-  def add(subject, device) when subject == "devices" do
-    with :ok <- GenServer.call(wall_pid(), {:add_device, Devices.setting_projection(device)}),
-         :ok <- GenServer.call(vpn_pid(), {:set_config, Devices.to_peer_list()}) do
+  def add(%Device{} = device) do
+    if :ok ==
+         send_event(get_gateway_id(), "add_peer", Devices.to_peer(device)) do
       :ok
     else
-      _err ->
-        Notifications.add(%{
-          type: :error,
-          message: """
-          #{device.name} was created successfully but an error occurred applying its
-          configuration to the WireGuard interface. Check the logs for more
-          information.
-          """,
-          timestamp: DateTime.utc_now(),
-          user: Users.get_user!(device.user_id).email
-        })
+      Notifications.add(%{
+        type: :error,
+        message: """
+        #{device.name} was created successfully but an error occurred applying its
+        configuration to the WireGuard interface. Check the logs for more
+        information.
+        """,
+        timestamp: DateTime.utc_now(),
+        user: Users.get_user!(device.user_id).email
+      })
```

Comment:
Probably better to add an `add_error/2` function to the Notifications module, then you can just do:

```suggestion
      Notifications.add_error(:apply_changes, device)
```